### ISSUE TITLE
Trained Model and Code? 

### ISSUE THREAD

**Issue by ysong10** [morawi/FashionColor-0#2]
Hi,
Thank you for providing the code corresponding to the paper; the outcomes are noteworthy, especially when compared with TinEye. I have observed that there is some code missing, as highlighted in a separate issue. Could you also share the trained model and the associated code, please?

**Comment by morawi** [comment]
Hi 

the code in here is outdated .... there's the another repo that has even more functionalities ... 
it has a better color  counting methods which I took some time to discover ... as well as some other functions ... have a look  and let me know what you think .... 
[FashionColor](https://github.com/morawi/FashionColor)

**Comment by ysong10** [comment]
Hi,
Thank you so much for sharing the code! I will check it out. I have another thing to clarify with you, so from the paper, it seems you fine-tuned one pretrained masked-rcnn model using that 50k modanet dataset, is this correct?

**Comment by dshaw002** [comment]
@ysong10 I've found a pre-trained model here: https://github.com/cad0p/maskrcnn-modanet/blob/38551c4ae684000a4c6f31c46c6912db5fdbe2c3/maskrcnn_modanet/download.sh#L141

EDIT: Looks like you can train model using get_prediction_method from his custom segmentation functions: https://github.com/morawi/FashionColor/blob/174ea1d83af48a4afc48e564b3440e42ef44de40/draw_image.py#L28

You can see something more detailed in one of the pytorch documents he references: https://pytorch.org/tutorials/intermediate/torchvision_tutorial.html

From there, I'm assuming you can train the saved model to the pth file using pickle, etc. I'm also going through this on my own for a project and will post an article when done

**Comment by nadia886** [comment]
> @ysong10 I've found a pre-trained model here: https://github.com/cad0p/maskrcnn-modanet/blob/38551c4ae684000a4c6f31c46c6912db5fdbe2c3/maskrcnn_modanet/download.sh#L141
> 
> EDIT: Looks like you can train model using get_prediction_method from his custom segmentation functions: https://github.com/morawi/FashionColor/blob/174ea1d83af48a4afc48e564b3440e42ef44de40/draw_image.py#L28
> 
> You can see something more detailed in one of the pytorch documents he references: https://pytorch.org/tutorials/intermediate/torchvision_tutorial.html
> 
> From there, I'm assuming you can train the saved model to the pth file using pickle, etc. I'm also going through this on my own for a project and will post an article when done

Hi, do you know how can I use the trained model in google colab? 

Thanks for your help.

**Comment by dshaw002** [comment]
@nadia886  @ysong10 
I'll make my notebook open. I don't have the original 50k dataset, but found a much smaller subset here:

https://universe.roboflow.com/gizem/modanet-97ezb

I'm using YoLo to train it which seems more up-to-date: https://colab.research.google.com/drive/1lSO8p7i-oYaXUpouJD-XnyN9qTmi_Oh7?usp=sharing

Link to the pre-trained model here (YoLo v11 only):
https://drive.google.com/file/d/1Ny0ZFOvw8YTq1xXYOujvGzH9S2P07Hkz/view?usp=sharing

One important note is that I used vast.ai to train the model instead of Google Colab. The processor for Google Colab is quite slow. I wound up spending about $1 to rent an RTX 4090 for an hour, though it took like 10 minutes to train this small dataset. There are other larger datasets I will train on once I get the other pieces of my personal project working.

**Comment by dshaw002** [comment]
I put a version of the 50k maskrcnn-modanet dataset on Roboflow with annotations!

https://universe.roboflow.com/tsty-9lkwi/modanet-maskrcnn/dataset/1

Training it now. Will edit this comment with the trained model file. The above version model wasn't that good, but MaskRCNN or SAM would be more accurate I'm assuming. 

Weights here for 50k dataset trained for YoloV11 Segmentation on 100 epochs: https://drive.google.com/file/d/1cOSvFpKqLxpmkWYS7MTLldVLniTCsGZ-/view?usp=sharing

Fairly accurate (example): https://drive.google.com/file/d/1CFit1BPIfGNIF-65oD2YAF6cY-7h1V3Q/view?usp=sharing

You can use my notebook link above, download the packages in one of the first cells, upload the weights in the drive I shared in this comment and run the predict segment command close to the end
